### PR TITLE
#114 Fail if no generator regulates the main connex component

### LINKED ISSUE #114: Check the existence of a generator regulating the voltage
@marcochiaramello @cossartque I would be in favor of adding such a check in the launcher. What is your opinion?

### PR DESCRIPTION
**Pull request by lecourtoisflo**
closes #114

### CODE REVIEW

**Review by marcochiaramello — APPROVED** [review]

File: etc/DFLMessages_en_GB.dic [hunk 1]
```diff
@@ -43,6 +43,7 @@ ConnexityError                =     Slack node of id %1% not present in main con
 ConnexityErrorReCompute       =     Slack node of id %1% not present in main connex component: compute slack node only in main connex component
 SimulateInfo                  =     Initializing %1% simulation
 NetworkFileNotFound           =     Network file %1% does not exist
+NetworkHasNoGenerator         =     Network file %1% has no generator regulating voltage in the main connex component
```
Comment (cossartque) [inline etc/DFLMessages_en_GB.dic:46]: NetworkHasNoRegulatingGenerator ?

File: tests/main/CMakeLists.txt [hunk 2]
```diff
@@ -71,3 +71,7 @@ DEFINE_LAUNCH_TEST(distant_regulation)
 DEFINE_LAUNCH_TEST(special_characters)
 
 DEFINE_LAUNCH_TEST(phase_shifter)
+
+DEFINE_LAUNCH_TEST(launch_fail_generators)
+## Expected to fail because no generator regulates
```
Comment (cossartque) [inline tests/main/CMakeLists.txt:76]: regulates the voltage in the main connected component

File: tests/main/res/TestIIDM_launch_fail_generators.iidm [hunk 3]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="ISO-8859-1" standalone="no"?>
+<!--
+    Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
```
Comment (cossartque) [inline tests/main/res/TestIIDM_launch_fail_generators.iidm:3]: 2021

**Review by cossartque — CHANGES_REQUESTED** [review]

File: etc/DFLMessages_en_GB.dic (hunk 1 above)
Comment (lecourtoisflo) [reply etc/DFLMessages_en_GB.dic:46]: OK

File: tests/main/CMakeLists.txt (hunk 2 above)
Comment (lecourtoisflo) [reply tests/main/CMakeLists.txt:76]: OK

File: tests/main/res/TestIIDM_launch_fail_generators.iidm (hunk 3 above)
Comment (lecourtoisflo) [reply tests/main/res/TestIIDM_launch_fail_generators.iidm:3]: OK

**Review by cossartque — APPROVED** [review]

**Pull request merged**